src/Chapter3.hs

```diff
@@ -910,6 +1030,26 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = MkGold
+  { unGold :: Int
+  } deriving (Show)
+
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append g1 g2 = MkGold $ unGold g1 + unGold g2
+
+instance Append [a] where
+  append :: [a] -> [a] -> [a]
+  append l1 l2 = l1 ++ l2
+
+instance (Append a) => Append (Maybe a) where
+  append :: Maybe a -> Maybe a -> Maybe a
+  append Nothing r = r
+  append l Nothing = l
+  append (Just l) (Just r) = Just (append l r)
+
+doAppend :: (Append a) => a -> a -> a
+doAppend l r = append l r
```

Comment:
I'm just curious about reasons for implementing this function :slightly_smiling_face: